wgpu/base.py

```diff
@@ -63,13 +63,9 @@ def label(self):
         """
         return self._label
 
-
-class DictLike:
-    def __getitem__(self, name):
-        return self.__dict__[name]
-
-    def get(self, key, default=None):
-        return self.__dict__.get(key, default)
+    def __del__(self):
+        if hasattr(self, "destroy"):
+            self.destroy()
```

Comment:
I thought this was only invoked when using the `del` keyword, but apparently also by the garbage collector? That's good to know!

Reply:
Yes! The only thing is that there is no way to know *when* it gets called, though for variables in a function scope, not referenced from elsewhere, *when* is usually directly when the function returns.